scaaml/capture/ascad/ascad_capture.py

```diff
@@ -0,0 +1,274 @@
+"""Capture script for easier manipulation."""
+from pathlib import Path
+from typing import Dict, Literal
+
+from scaaml.aes_forward import AESSBOX
+from scaaml.io import Dataset
+from scaaml.capture.aes.capture_runner import CaptureRunner
+from scaaml.capture.aes.crypto_alg import SCryptoAlgorithm
+from scaaml.capture.aes.communication import CWCommunication
+from scaaml.capture.ascad.control import CWControl
+from scaaml.capture.ascad.scope import PicoScope, CWScope
+
+
+def capture_ascad_dataset(
+        firmware_sha256: str,
+        architecture: str,
+        implementation: str,
+        shortname: str,
+        description: str,
+        capture_info: Dict,
+        chip_id: int,
+        crypto_implementation=AESSBOX,
+        algorithm: str = 'simpleserial-aes',
+        version: int = 1,
+        root_path: str = '/mnt/storage/chipwhisperer',
+        url: str = '',
+        firmware_url: str = '',
+        paper_url: str = '',
+        licence: str = "https://creativecommons.org/licenses/by/4.0/",
+        examples_per_shard: int = 64,
+        measurements_info=None,
+        repetitions: int = 1,
+        train_keys: int = 3 * 1024,
+        train_plaintexts: int = 256,
+        holdout_keys: int = 0 * 1024,
+        holdout_plaintexts: int = 1) -> None:
+    """Capture or continue capturing the dataset.
+
+    Args:
+      firmware_sha256: Hash of the used firmware binary.
```

Comment:
nit: IntelHEX file. Because we also have ELF files but these are not reproducible from a compilation to another, yielding different hashes while the HEX files are stable.